RecoHGCal/TICL/plugins/TrackstersFromSimClustersProducer.cc

```diff
@@ -0,0 +1,151 @@
+// Author: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 02/2021
+
+// user include files
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+
+#include "DataFormats/HGCalReco/interface/Trackster.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "SimDataFormats/Associations/interface/LayerClusterToSimClusterAssociator.h"
+#include "SimDataFormats/CaloAnalysis/interface/SimCluster.h"
+#include "RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h"
+
+#include "TrackstersPCA.h"
+#include <vector>
+#include <iterator>
+#include <algorithm>
+using namespace ticl;
+
+namespace {
+  Trackster::ParticleType tracksterParticleTypeFromPdgId(int pdgId, int charge) {
+    if (pdgId == 111) {
```

Comment:
Why neutral pion is special here, with separate if? What will wrong if we have -111 somehow, will you count it as neutral hadron?

I assume you try to split the particle by PDG ID only in this if-else.

Reply:
neutral pions are not treated the same way as neutral hadrons. They are treated as electromagnetic objects for their decay in two photons

Reply:
I mean why it needs an "if" for this pi-0 and then you have else for others (with abs). I may miss the reason as I understand that you only define what should be which from PDG only in this if-else.

Reply:
I think I did it like this because there is no -111 particle